arrow/arrow.py

```diff
@@ -1285,12 +1291,25 @@ def gather_timeframes(_delta: float, _frame: TimeFrameLiteral) -> float:
                 for frame in frames:
                     delta = gather_timeframes(delta, frame)
 
-                if len(timeframes) < len(granularity):
+                if len(timeframes) < len(granularity) and not dynamic:
                     raise ValueError(
                         "Invalid level of granularity. "
                         "Please select between 'second', 'minute', 'hour', 'day', 'week', 'month' or 'year'."
                     )
 
+                # Needed to see if there are no units output an error
+                if not timeframes and dynamic:
+                    raise ValueError(
+                        "All provided granulairty values produced an output of zero. "
+                        "Consider using smaller granularities, or set the dynamic flag to False. "
+                    )
```

Comment:
How about defaulting to "just now" rather than raising an exception? If you imagine this being used with user input, it would pretty much be a requirement to always wrap it in a try-except due to this being raised.

Reply:
Also, you misspelled granularity.

Reply:
A user could have a delta of let's say 2 days but only have the granularity of ["year, "month", "week"].  If they had dynamic on, it would output "just now." I think it would be a better idea to error out, then to give an inaccurate answer in that scenario.

Reply:
Ah, it doesn't convert in that case e.g. 2 days = 2/7ths of a week? That's a good point then.

Will this still raise the exception if all granularities are provided, but all values are 0, or will it actually display "just now" in that case? I think it should be able to do that.

Reply:
In fact, it could show "just now" if all units evaluate to zero, regardless of how which granularities are provided, not just if all are provided.

Reply:
Though that would be inconsistent with the behaviour of e.g. `granularity="year"` returning `'0 years'` rather than `'just now'`. On the other hand, it seems like it raising an error in those cases should be avoidable somehow.

Reply:
Our main goal is that if you provide a granularity, you expect the output to contain said granularity (with the exception of the omit zeros/dynamic functionality). Trying to figure out whether we should or shouldn't adapt the output to include some other unit seems unnecessary when we already have the auto function in humanize.

Reply:
Raising an error absolves arrow of having to make the tough decision of how to handle this. Letting the user handle the error gives them flexibility, but not all users might prefer that at the cost of having to practically always handle this error if they're dealing with unknown inputs. 

It's a matter of which use case is more common: wanting custom behaviour to handle this edge case, or wanting to not have to think about it. Either way, the user can anticipate what the result will be by subtracting the times and manually inspecting the delta before calling `humanize`. If they see the delta will result in all zeros, they can handle it instead of relying on the default behaviour proposed below. Of course, that's not as convenient as just catching an exception, but I don't see a way to make both sides happy.

---

The most consistent solution may be to return zero in the smallest unit of the given granularity. This would ensure that while `dynamic=True` may omit some units in the given granularity, it will never introduce new units. Consider

```py
>>> a = arrow.get(2021, 8, 8)
>>> b = arrow.get(2021, 8, 10)
>>> a.humanize(b, granularity=["year", "month", "week"])
'0 years 0 months and 0 weeks ago'
```

It has no problem omitting the "2 days" even though it's the only non-zero unit. This is arguably not very useful, but it's what the current behaviour is. There are probably use cases that need to strictly follow the granularity, and those users appreciate this behaviour. Anyway, following from this behaviour, it should then also be acceptable for this to happen

```py
>>> a = arrow.get(2021, 8, 8)
>>> b = arrow.get(2021, 8, 10)
>>> a.humanize(b, granularity=["year", "month", "week"], dynamic=True)
'0 weeks ago'
```

If the user has dynamic on, that is an expression of an intent to cut down on the zeros in the output. I'd say it's more practical to make a compromise to return 1 zero than to take a strict stance of "must have no zeros" and be forced to raise an exception.

Reply:
We can only do so much, arrow in the end is a library meant to work together with the dev, not think or do things the dev might not be aware of and does behind the scenes without awareness and not raise an exception that the dev might even be expecting to see raised.